src/elysium/ecdsa_signature.cpp

```diff
@@ -17,32 +17,39 @@ ECDSASignature::ECDSASignature(secp256k1_ecdsa_signature const &sig)
 {
 }
 
-ECDSASignature::ECDSASignature(ECDSAContext const &context, unsigned char const *signature, size_t len)
-    : valid(false)
+ECDSASignature ECDSASignature::Parse(ECDSAContext const &context, unsigned char const *signature, size_t len)
 {
+    ECDSASignature sig;
+
     if (len >= 70 && len <= 72) {
         if (1 == secp256k1_ecdsa_signature_parse_der(
             context.Get(),
-            &(this->signature),
+            &(sig.signature),
             signature,
             len))
         {
-            valid = true;
+            sig.valid = true;
```

Comment:
I think we should throw instead. It looks weird that this method can return an invalid signature.